src/bech32.c

```diff
@@ -206,30 +227,34 @@ int wally_addr_segwit_from_bytes(const unsigned char *bytes, size_t bytes_len,
     char result[90];
     size_t push_size;
     int ret;
+    uint8_t witver;
 
     if (output)
         *output = 0;
 
     if (!addr_family || flags || !bytes || !bytes_len || !output)
         return WALLY_EINVAL;
 
-    if (bytes[0] != 0)
-        return WALLY_EINVAL; /* Only v0 witness programs are currently allowed */
+    witver = bytes[0];
+    if (witver != 0)
+        witver -= 0x50;
```

Comment:
This needs some explanation, why are we subtracting 0x50 here?

Reply:
OP_0 is 0x0, and OP_1 .. OP_16 is 0x51 to 0x60. So, if it is not OP_0, then we can get the version by subtracting 0x50 from the value.

Reply:
If you go that route, adding a comment would help. But you'd also have to check that `witver` > `0x50` to prevent underflow.

It'll be years before there's a v2 witness version, so just checking explictly for `OP_0` and `OP_1` might be easier.

Reply:
This is what I meant above, we should just be checking for the allowed versions OP_0 and OP_1 (disregarding the core advice to allow unknown versions since as v0 to v1 showed us, there is no guarantee this will work without further changes).

Reply:
@Sjors The next line there is a check that will prevent any underflow to pass.
```
if (witver > 1) /* For now we only support v0 and v1 addresses */
        return WALLY_EINVAL;
```
In fact, it will only allow the exact value of witver that is in {0,1}. As the passed byte is `uchar` and the witver is taken as `uint8_t` this essentially guarantees that only OP_0 and OP_1 will be passed. I wanted to keep this code instead of exactly checking OP_0 and OP_1 because in the future if OP_2 needs to be supported and it uses bech32m then all we need to do will be to change the comparison line.
But if you and @jgriffiths think it is better to check OP_0 and OP_1 explicitly I will change it to that. Let me know.

Reply:
I prefer the explicit comparison because it avoids magic numbers in the code. Also its likely that when the next version is added we may have further logic to check after verifying the version, so keeping the version check a single 'if' in the future may well not be possible anyway.

Reply:
@jgriffiths I have added explicit comparison. Kindly review the patch.

Reply:
Thanks @fadedreamz - I am a little caught up with some gdk changes so it might be a while before I can look at this, please know it's on my list to action!